database/tables/instance_questions.pg

```diff
@@ -13,6 +13,8 @@ columns
     incremental_submission_points_array: double precision[]
     incremental_submission_score_array: double precision[]
     last_submission_score: double precision
+    manual_grading_conflict: boolean default false
```

Comment:
In tracing through the use of `manual_grading_conflict` and `grading_job_conflict` in the rest of the code, it looks to me like it's only protecting against the case where there are exactly _two_ users grading a question at once, where the first user graded and the second user has a conflict. It doesn't seem to handle the case of three (or more) simultaneous graders or two graders who are both trying to repeatedly grade.

One way I see this is that we are storing a single `manual_grading_conflict` flag on the instance question. Consider the case where:
1. User A starts grading the question
2. User B starts grading the question
3. User C starts grading the question
4. User A submits a grade
5. User B tries to submit a grade, gets a conflict
6. User C tries to submit a grade, but there is already a conflict -- now what?

Rather than storing a `conflict` flag on the instance question, I would prefer to store it on the `grading_job` so that multiple users can have separate conflicting grading jobs at the same time.

Or am I misunderstanding the logic here?

Reply:
I think this is a valid point.

Current implementation: If user C submits a grade, but there is already a conflict from user A and B, then user C should be presented with the last two grading jobs to choose from. As the view only supports loading two conflicts, user A's grading job disappears from the options. This is not ideal.

Possible implementation: Store the grading_job conflict reference in the grading_job. Hence, user B's grading_job points to user A's. User C's grading_job points to user B's. Each grading_job conflict would load sequentially, in the worst case scenario that both users B and C abandon their work after the conflicts are created, as `grading_job` conflicts count for as ungraded on the Assessment Manual Grading page.

![image](https://user-images.githubusercontent.com/7872295/113744103-3fe4af80-96b9-11eb-9592-4b4e83258532.png)

Once the last manual grading conflict is resolved, we can do the grade calculation.